### PR TITLE
Advancedstats

### PR DESCRIPTION
**Pull request by Jean-lucLanglois**
Added the Advanced Stats Module

### CODE REVIEW

File: .ruby-version [hunk 1]
```diff
@@ -0,0 +1 @@
+2.3.0
```
Comment (dblock) [inline .ruby-version]: We shouldn't need this.

File: lib/iex/resources/advanced_stats.rb [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+      property 'price_to_sales', from: 'priceToSales'
+      property 'price_to_sales_dollar', from: 'priceToSales', with: ->(v) { Resource.to_dollar(amount: v) }
+      property 'price_to_book', from: 'priceToBook'
+      property 'forward_pe_ratio', from: 'forwardPERatio'
+      property 'avg_30_volume', from: 'avg30Volume'
+      property 'peg_ratio', from: 'pegRatio'
+      property 'beta'
+ 
+      def initialize(data = {})
+        super
```
Comment (dblock) [inline lib/iex/resources/advanced_stats.rb]: Can be deleted since it's just a call to `super`?

**Comment by Jean-lucLanglois** [comment]
The behavior of AdvancedStats on IEX is to duplicate the calls of the KeyStats module and adding another ~27 or so call from totalCash to peLow. Do we want the module to duplicate the KeyStats call as well as the Advanced call or just the ~27 or so advanced calls excluding the KeyStats calls? Thank you for your help.

File: .ruby-version (hunk 1 above)
Comment (Jean-lucLanglois) [reply .ruby-version]: I added the file to .gitignore

File: lib/iex/resources/advanced_stats.rb (hunk 2 above)
Comment (Jean-lucLanglois) [reply lib/iex/resources/advanced_stats.rb]: deleted the call to super

**Comment by dblock** [comment]
> The behavior of AdvancedStats on IEX is to duplicate the calls of the KeyStats module and adding another ~27 or so call from totalCash to peLow. Do we want the module to duplicate the KeyStats call as well as the Advanced call or just the ~27 or so advanced calls excluding the KeyStats calls? Thank you for your help.

I think you can try to inherit one from the other? We do this in charts.

**Comment by dblock** [comment]
I'll wait till this gets to 🍏 to CR the whole thing. LMK if you need help.

**Comment by Jean-lucLanglois** [comment]
> > The behavior of AdvancedStats on IEX is to duplicate the calls of the KeyStats module and adding another ~27 or so call from totalCash to peLow. Do we want the module to duplicate the KeyStats call as well as the Advanced call or just the ~27 or so advanced calls excluding the KeyStats calls? Thank you for your help.
> 
> I think you can try to inherit one from the other? We do this in charts.

Can you point me in the right direction. I'm not really getting how to do this. Thank you

**Comment by dblock** [comment]
> > I think you can try to inherit one from the other? We do this in charts.
> 
> Can you point me in the right direction. I'm not really getting how to do this. Thank you

Don't worry about it for this CR for now. Get this to a passing build?

**Comment by dblock** [comment]
For rubocop run `rubocop -a ; rubocop --auto-gen-config`.

**Comment by dblock** [comment]
Do you want help with this @Jean-lucLanglois ? I can take it over and merge. Or if you are ok to hang in there I can help you work through it. Looks like your rubocop is making a mess, it's not reading `.rubocop.yml` and throwing all the `# frozen_string_literal: true` around. Not sure why. I would undo the last commit and retry, and iterate locally until `rake` passes without errors`.

**Comment by Jean-lucLanglois** [comment]
thanks @dblock, I don't mind figuring it out. here is my output for rubocop -d
```
rubocop -d
For /Users/jean-luc/Sites/iex-ruby-client: configuration from /Users/jean-luc/Sites/iex-ruby-client/.rubocop.yml
Inheriting configuration from /Users/jean-luc/Sites/iex-ruby-client/.rubocop_todo.yml
Default configuration from /Users/jean-luc/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/rubocop-0.72.0/config/default.yml
```

**Comment by Jean-lucLanglois** [comment]
I'm running the latest version of rubocop
```
rubocop -V
0.72.0 (using Parser 2.6.3.0, running on ruby 2.3.0 x86_64-darwin16)
```

**Comment by Jean-lucLanglois** [comment]
I reset back to the last commit and I pushed the branch. I noticed that Travis CI is also not reading the rubocop.yml file and failing on '# frozen_string_literal: true.'

**Comment by Jean-lucLanglois** [comment]
@dblock, I have rubocop running clean. I needed to update the .rubocop_todo.yml to version 0.72.0. 
I now get error 
`'IEX::Errors::PermissionDeniedError:The API key provided is not valid.'`
not sure how to proceed from there. Could you assist me in getting this resolved?

File: .ruby-version (hunk 1 above)
Comment (dblock) [reply .ruby-version]: Since it's in an older commit you should also `git rm` it.

File: spec/fixtures/iex/advanced_stats/invalid.yml [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+      Access-Control-Allow-Headers:
+      - Origin, X-Requested-With, Content-Type, Accept
+    body:
+      encoding: ASCII-8BIT
+      string: Unknown symbol
+    http_version: 
+  recorded_at: Wed, 19 Jun 2019 19:05:36 GMT
+- request:
+    method: get
+    uri: https://cloud.iexapis.com/v1/stock/INVALID/advanced-stats?token=pk_7c131cc2187b4c5583fc16816b5cf05e
```
Comment (dblock) [inline spec/fixtures/iex/advanced_stats/invalid.yml]: This K7 leaked your real token!

**Comment by dblock** [comment]
The IEX client is failing to make an API call complaining about the key. This is saying that it's trying to talk to iexcloud.com instead of using a recorded cassette by VCR. Aka that the recording .yml doesn't match what it's trying to do.

Usually when I add an API I will run the spec once with `IEX_API_PUBLISHABLE_TOKEN=<real token> rspec ...` and that will create a yml recording. Then I edit it and swap the token for `test-iex-api-publishable-token` manually in that recording. That's what we use by default [here](https://github.com/dblock/iex-ruby-client/blob/b2cf66b7e7f57cd19522767594dd734ed171f8d1/spec/spec_helper.rb#L12). So next time run tests without that env variable set and it should not try to make any real calls to IEX and use the VCR .yml. I then `git add` it and make sure no more changes are added to the k7.

This is a little cumbersome but I didn't find a better way.

I think you should delete the .yml's that you have right now for the APIs you have added and try to do the above.

**Comment by FanaHOVA** [comment]
@rodolfobandeira @dblock just ran across this PR, adding those changes and cleaning some stuff up (like using snake case instead of camel case) and getting this feature to a mergeable state since this one seems stale.

**Comment by FanaHOVA** [comment]
See #84.

**Comment by dblock** [comment]
Closed via https://github.com/dblock/iex-ruby-client/pull/84

**Pull request closed without merging**